reactapp/test/main/main.w

```diff
@@ -0,0 +1,20 @@
+bring cloud;
+
+bring "../../lib.w" as reactapp;
+
+let api = new cloud.Api(cors: true);
+
+api.get("/", inflight () => {
+  return {
+    status: 200,
+    body: "Hello World! API",
+  };
+});
+
+let project = new reactapp.ReactApp(
+  projectPath: "../react-project",
+  localPort: 4500,
+);
+
+project.addEnvironment("API_URL", api.url);
+project.addEnvironment("TEXT", "Hello World!");
```

Comment:
Maybe we can add a test, like:
```suggestion
project.addEnvironment("TEXT", "Hello World!");

if util.env("WING_TARGET") == "sim" {
  test "we can access the app" {
    let url = project.url;
  
    util.waitUntil(inflight () => {
      try {
        let response = http.get(url);
        return response.status == 200;
      } catch e {
        return false;
      }
    }, timeout: 10s);
  
    let response = http.get(url);
    expect.equal(response.status, 200);
    assert(response.body.startsWith("<!DOCTYPE html>"));
  }
}


```

Reply:
I need to run `npm install` in the `react-project` dir before the test starts running, but when I add a new step in `reactapp-pull.yaml`:

```yaml
      - name: Install React project dependencies
        run: npm install
        working-directory: reactapp/test/react-project
```

the Pull Request Diff workflow failed :(